addons/web_editor/static/lib/summernote/src/js/module/Editor.js

```diff
@@ -648,11 +650,18 @@ define([
         $(anchor).attr('href', linkUrl);
         $(anchor).attr('class', linkInfo.className || null); // ODOO: addition
         $(anchor).css(linkInfo.style || {}); // ODOO: addition
-        if (isNewWindow) {
+        if (isNewWindow && !isModalTarget) {
           $(anchor).attr('target', '_blank');
         } else {
           $(anchor).removeAttr('target');
         }
+        if (isModalTarget) {
+          $(anchor)[0].dataset.toggle = 'modal';
+          $(anchor)[0].dataset.target = selector;
+        } else if ($(anchor)[0].dataset.toggle = 'modal') {
+          delete $(anchor)[0].dataset.toggle;
+          delete $(anchor)[0].dataset.target;
+        }
```

Comment:
Modification of the lib file ... this can't be good :/ Especially since summernote will be dropped by the new editor. Why would we need this? What is done here seems really the job of the link dialog

Reply:
I haven't found a way to move 'data-attribute' (data-modal, data-target) from the link dialog to the editor. The only data that is handled are style, href and classes. I don't see how i can do it in an other way

Reply:
I don't understand, is not the link dialog in charge of changing href & co? Why can't it change data attributes too?

Reply:
As I understand, the link dialog pick up the information from the modal fields and give those information to the editor...
And then the editor build the link

Reply:
Dammit, what a mess :grimacing: Ok then but don't modify the lib file, you can override that `createLink` method in our own files for this case instead of modifying it directly in the lib file. Just like we override `showLinkDialog`.

Also, it would be great if you can add a test tour which drops a modal, create a link to that modal, save, click on the link and checks the modal is opened. We'll need better testing tools in the future but we should start making better test coverage especially for this kind of things that will simply be lost by the editor team if we don't make a test for it :wink: